spec.bs

```diff
@@ -3720,42 +3757,57 @@ methods for event-level <dfn>forDebuggingOnly reports</dfn> for winning and losi
+            [=bid debug reporting info/top level seller debug win report url=], |seller|, |fromServer|,
+            and |auctionReportInfo|'s [=auction report info/debug win report urls=].
       1. Otherwise:
         1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
           [=bid debug reporting info/bidder debug loss report url=], |bidDebugReportInfo|'s
-          [=bid debug reporting info/interest group owner=], and |auctionReportInfo|'s
+          [=bid debug reporting info/interest group owner=], |fromServer|, and |auctionReportInfo|'s
           [=auction report info/debug loss report urls=].
         1. If |bidDebugReportInfo|'s [=bid debug reporting info/component seller=] is null:
           1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
-            [=bid debug reporting info/seller debug loss report url=], |seller|, and
+            [=bid debug reporting info/seller debug loss report url=], |seller|, |fromServer|, and
             |auctionReportInfo|'s [=auction report info/debug loss report urls=].
         1. Otherwise:
           1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
             [=bid debug reporting info/seller debug loss report url=], |bidDebugReportInfo|'s
-            [=bid debug reporting info/component seller=], |auctionReportInfo|'s
+            [=bid debug reporting info/component seller=], |fromServer|, |auctionReportInfo|'s
             [=auction report info/debug loss report urls=].
           1. [=Collect a single forDebuggingOnly report=] with |bidDebugReportInfo|'s
-            [=bid debug reporting info/top level seller debug loss report url=], |seller|, and
-            |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+            [=bid debug reporting info/top level seller debug loss report url=], |seller|,
+            |fromServer|, and |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+      1. [=map/For each=] |invokingOrigin| → |debugURLs| of |bidDebugReportInfo|'s
+        [=bid debug reporting info/server filtered debugging only reports=]:
+        1. If |debugURLs| [=list/is empty=]:
+          1. If the result of running [=is debugging only in cooldown or lockout=] with
+            |invokingOrigin| is false, then [=update debug report cooldown=] with |invokingOrigin|.
+          1. [=iteration/Continue=].
+        1. [=list/For each=] |url| of |debugURLs|:
+          1. [=Collect a single forDebuggingOnly report=] with |url|, |invokingOrigin|,
+            |fromServer| (true), and |auctionReportInfo|'s [=auction report info/debug loss report urls=].
+
+            Note: For server filtered ones, post auction signals were filled on the server side. As
+              a result, there's no difference if they go to loss or win lists, since post auction
+              signals was the difference.

```

Comment:
I don't think this is true, the win urls don't get sent if there is no winner:
https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/1325.html#ref-for-auction-report-info-debug-loss-report-urls:~:text=Otherwise%20if%20winnerInfo%20is%20null%20or%20winnerInfo%E2%80%99s%20leading%20bid%20is%20null%3A

...so I think this is doing the right thing with the wrong explanation.